library/Icingadb/Common/Auth.php

```diff
@@ -321,4 +327,38 @@ protected function parseBlacklist(string $blacklist, string $column): Filter\Non
 
         return $filter;
     }
+
+    /**
+     * Force query optimization on the given service/host filter rule
+     *
+     * Applies forceOptimization, when the given filter contains one of those columns(hostgroup.name,servicegroup.name)
+     *
+     * @param Filter\Rule $filterRule
+     * @param string $filterColumn
+     *
+     * @return void
+     */
+    protected function forceQueryOptimization(Filter\Rule $filterRule, string $filterColumn)
+    {
+        if ($filterColumn !== 'hostgroup.name' && $filterColumn !== 'servicegroup.name') {
+            return;
+        }
```

Comment:
It's protected, and only used explicitly with hardcoded parameters. I don't think we need this.